### ISSUE TITLE
Syntax error: function/subroutine name same as fortran keyword

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#191]
If we use a fortran keyword as function/subroutine name, the front-end will produce a syntax error. This is permitted by the FORTRAN standard. 

Code example: 
```fortran
MODULE mod1
CONTAINS
  LOGICAL FUNCTION SELECT(a, b)
    INTEGER :: a, b
    SELECT = .TRUE.
  END FUNCTION
END MODULE mod1
```

```bash
"mod1.f90:mod1", line 4: syntax error
"mod1.f90:mod1", line 5: only function/subroutine statement are allowed in contains top level
"mod1.f90:mod1", line 6: only function/subroutine statement are allowed in contains top level
"mod1.f90:mod1", line 10: declaration among executables
```

As mentioned in the FORTRAN Standard: 
`2.5.2 Statement keyword`
```
A statement keyword is not a reserved word; that is, a name with the same spelling is allowed. In the
syntax rules, such keywords appear literally. In descriptive text, this meaning is denoted by the term
“keyword” without any modifier. Examples of statement keywords are IF, READ, UNIT, KIND, and INTEGER.
```

**Comment by clementval** [comment]
@h-murai @shingo-s This is probably very tricky to allow FORTRAN `keyword` also as `identifier` for function, variable ...

**Comment by clementval** [comment]
@h-murai I found the problem and a fix is ready in PR #195

**Comment by clementval** [comment]
Closed with #195
